### ISSUE TITLE
Block Editor: Consider removing the contrast checker when block has only links

### ISSUE THREAD

**Issue by gziolo** [WordPress/gutenberg#37549]
### Description

Discovered this while testing the Comment Reply Link and Comment Edit Link blocks. Both blocks don't contain any regular text because everything is wrapped with a link. It also means that block supports have color controls enabled only for the background and the link. Initially reported in https://github.com/WordPress/gutenberg/pull/35774#issuecomment-952046182

> Background color + link color trigger contrast checker warning when they shouldn't:
> 
> <img width="279" alt="Screen Shot 2021-10-26 at 16 58 49" src="https://user-images.githubusercontent.com/699132/138908973-7b03c82b-040b-4050-aba6-d8a5d380c5a2.png">
> 
> I think the checker compares the background color selected and the default text color read from the same `div`, but we don't care about it at all.

### Step-by-step reproduction instructions

1. Insert the Comments Query Loop block into a template or post/page.
2. Start editing the Comment Reply Link or the Comment Edit Link block.
3. Open the settings sidebar and go to the colors panel.
4. Set the background color to black and the link color to white which is the best possible contrast configuration 😄 
5. The contrast checker will show the warning that the background color and the text color (I assume it's set to black in the theme) needs to be updated to increase the contrast. 

### Screenshots, screen recording, code snippet

The up to date version looks this way:

<img width="1020" alt="Screenshot 2021-12-21 at 09 47 32" src="https://user-images.githubusercontent.com/699132/146919931-4a22d144-03ee-48ad-8be4-265e2a04c887.png">

### Environment info

- WordPress 5.9 beta 3
- Gutenberg 12.2 (actually the latest version from `trunk`)

The browser and OS don't matter here.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by gziolo** [comment]
@jorgefilipecosta, I would be curious to hear your feedback what would be the ideal solution for this edge case? As far as I can see in the codebase, there is no way to use block supports to disable the contrast checker.

I played a bit with a very naive approach where the contrast checker would be disabled when there is no block support enabled for the text color in this place:

https://github.com/WordPress/gutenberg/blob/3d52a8c68c90f130b7a629d67dcc167471a5b336/packages/block-editor/src/hooks/color.js#L377-L380

So the check would ensure that is only enabled when :

```js
Platform.OS === 'web' && ! gradient && ! style?.color?.gradient && hasTextColor && hasBackgroundColor
```

The question is whether we want to improve the experience and offer a contrast checker also for links vs background but in that case we would have to convert this issue into an enhancement.

**Comment by ivan-ottinger** [comment]
One observation related to this issue. In some circumstances, the contrast checker displays also when there's only a background color available on the block:

![Markup on 2022-02-10 at 16:05:05](https://user-images.githubusercontent.com/25105483/153435637-c05d1dda-ef7f-45a8-9686-a3623bfaca42.png)

This is a block that displays images in its content only - so it would be better if the checker wasn't displayed in this case.

**Comment by ramonjd** [comment]
Similar issue over at https://github.com/WordPress/gutenberg/pull/38428#discussion_r805468375 but related to the Separator Block. 

Excerpts from that discussion:

> The [Color Panel's implementation of the ContrastChecker](https://github.com/WordPress/gutenberg/blob/8675f86f885ca389f3bca39971b22cb897bab61b/packages/block-editor/src/hooks/color-panel.js#L74) is working as (originally) intended in that it attempts to compare the detected background color with any detected text color, irrespective of whether the selected block opts into color.text or not.

> So it's not about whether the component or the selected block has a color, background or otherwise, but whether the selected block's color choices are readable when checked against any detected text color or detected background color.

> In the case of the Separator block, even though it technically does not define a color or has any text, a color is nevertheless inherited from the has-background-color classname, and, if we remove that, from the body.

The Separator Block isn't really meant to house text, but the contrast checker detects an inherited `color` and tries anyway.

I think it would be helpful to completely disable (or enable) the contrast checker for specific blocks, possibly through `block.json`.

What do you think?
